okta-idx-kotlin/src/main/java/com/okta/idx/kotlin/dto/IdxCapability.kt

```diff
@@ -181,6 +184,85 @@ class IdxTotpCapability internal constructor(
     }
 }
 
+/**
+ * Describes the WebAuthn registration capability associated with an [IdxAuthenticator].
+ *
+ * @property activationData The JSON string containing the public key credential creation options for WebAuthn activation.
+ */
+class IdxWebAuthnRegistrationCapability internal constructor(
+    private val activationData: String,
+) : IdxAuthenticator.Capability {
+
+    /**
+     * Returns the public key credential creation options as a JSON string.
+     *
+     * @param rpId Optional relying party ID. If provided and not blank, overrides the default `rp.id` in the activation data if present.
+     * @return A [Result] containing the JSON string with the `rp.id` field set to [rpId] if specified, or the original activation data otherwise.
+     */
+    fun publicKeyCredentialCreationOptions(rpId: String? = null): Result<String> = runCatching {
+        val rootObject = JSONObject(activationData)
+        val rpObject = rootObject.getJSONObject("rp")
+
+        // Determine the final rpId value. If rpId is provided and not blank, use it; otherwise, check for an existing "id" in the rp object.
+        // If neither is available, check for "appid" in "u2fParams" and use its host as the rpId.
+        if (!rpId.isNullOrBlank()) {
+            rpObject.put("id", rpId)
+        } else if (!rpObject.has("id")) {
+            val derivedRpId = rpObject.optJSONObject("u2fParams")?.optString("appid")?.let { appId ->
+                try {
+                    URI(appId).host
+                } catch (_: URISyntaxException) {
+                    null
+                }
+            }
+            derivedRpId?.let {
+                rpObject.put("id", it)
+            }
+        }
+
+        rootObject.toString()
+    }
+}
+
+/**
+ * Describes the WebAuthn authentication capability associated with an [IdxAuthenticator].
+ *
+ * @property challengeData The JSON string containing the public key credential creation options for WebAuthn activation.
+ */
+class IdxWebAuthnAuthenticationCapability internal constructor(
+    private val _challengeData: String,
+) : IdxAuthenticator.Capability {
+
+    /**
+     * Returns the challengeData.
+     *
+     * @param rpId Optional relying party ID. If provided and not blank, overrides the default `rpId` in the challenge data if present.
+     * @return A [Result] containing the JSON string with the `rpId` field set to [rpId] if specified, or the original challenge data otherwise.
+     */
+    fun challengeData(rpId: String? = null): Result<String> = runCatching {
```

Comment:
Similar to registration, the Swift SDK names this [`relyingPartyIdentifier`](https://github.com/okta/okta-mobile-swift/blob/master/Sources/OktaIdxAuth/Capabilities/WebAuthnAuthenticationCapability.swift#L25).

Also, in the Swift SDK this is a mutable property on the capability itself, which allows for the default / server-supplied value to be provided here.  This way a developer can see what the original value was, or can override/replace it as needed.

This difference in patterns may be fine though.